### DISCUSSION TITLE
Is there a supported way to access exception events in spans?

### DISCUSSION THREAD

**Discussion by io7m** [open-telemetry/opentelemetry-java#6619] [Q&A]
Hello!

If I'm implementing the `SpanExporter` interface, then I'm going to be inspecting spans and the events within those spans. It's possible to access the contents of events via the public `EventData` interface. 

Unfortunately, if an event represents a recorded exception, then it'll be delivered using the `AutoValue_ImmutableExceptionEventData` class, which is descendant of the `ExceptionEventData` interface, which isn't public. There doesn't appear to be any other way to get at the exception data via a public API.

Am I missing something?

**Comment by jkwatson** [comment]
`ExceptionEventData` _is_ public, it's just in an `internal` package. I believe (@jack-berg might have better memory here) that we haven't stabilized that interface, as we have been waiting for feedback on whether it works for people. So, you are free to use `ExceptionEventData` for now, but it almost certainly will change at some point in the future, so you might need to be alert for any changes we might make to it (repackaging, in particular).

**Comment by io7m** [comment]
The JavaDoc says otherwise:

https://github.com/open-telemetry/opentelemetry-java/blob/6fc1d216ca0f199670a1cc5b081ad764d4a043fb/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/data/ExceptionEventData.java#L15

> This class is internal and is hence not for public use.

**Comment by io7m** [comment]
I suppose there's more than one definition of `public` here. It's public in the language access modifier sense, but it's not public in the sense that it's marked as internal and the documentation says not to use it. :slightly_smiling_face:

**Reply by jkwatson** [reply]
Yes. @jack-berg do you recall if we had plans to promote this to an official public API?

**Reply by jack-berg** [reply]
I think we've just waiting for feedback, positive or negative. Personally, its simple enough that I'm willing to go out on a limb and stabilize it. Waiting for feedback may be overkill.

**Reply by jack-berg** [reply]
Opened https://github.com/open-telemetry/opentelemetry-java/issues/6630 to track.
